### ISSUE TITLE
transmission-openvpn container is not working

### ISSUE THREAD

**Issue by rpforce** [haugene/docker-transmission-openvpn#1363]
So I am running this on OMV5 using portainer.
I added the following to try to get this installed and working but the transmission-openvpn just shows unhealthy.

version: '2.4'
services:
 transmission-openvpn:
    volumes:
        - /srv/dev-disk-by-label-media/downloads/:/data
        - /etc/localtime:/etc/localtime:ro
    environment:
        - PUID=998
        - PGID=100
        - CREATE_TUN_DEVICE=true
        - OPENVPN_PROVIDER=NORDVPN
        # - OPENVPN_CONFIG=default
        - NORDVPN_COUNTRY=US
        - NORDVPN_CATEGORY=legacy_p2p
        - NORDVPN_PROTOCOL=udp
        - OPENVPN_USERNAME=KZoU7PvyoyoJWYnrboY7SUWb
        - OPENVPN_PASSWORD=8HjsDTjwJxtFUWRiv2gCY19y
        - OPENVPN_OPTS=--inactive 3600 --ping 10 --ping-exit 60
        - WEBPROXY_ENABLED=false
        - LOCAL_NETWORK=192.168.1.109/24
        - TRANSMISSION_SCRAPE_PAUSED_TORRENTS_ENABLED=false
        - DNS=8.8.8.8,8.8.4.4
    cap_add:
        - NET_ADMIN
    logging:
        driver: json-file
        options:
            max-size: 10m
    ports:
        - 9091:9091
        - 9117:9117
        - 7878:7878
        - 8989:8989
        - 8686:8686
        - 5299:5299
    restart: always
    image: haugene/transmission-openvpn

 jackett:
    image: linuxserver/jackett
    container_name: jackett
    network_mode: "service:transmission-openvpn"
    environment:
        - PUID=998
        - PGID=100
        - TZ=America/Chicago
        - AUTO_UPDATE=true #optional
        - RUN_OPTS=#optional
    volumes:
        - /srv/dev-disk-by-label-data/appdata/jackett:/config
        - /srv/dev-disk-by-label-data/downloads/watch:/downloads
    restart: unless-stopped

 radarr:
    image: linuxserver/radarr
    network_mode: "service:transmission-openvpn"
    container_name: radarr
    environment:
        - PUID=998
        - PGID=100
        - TZ=America/Chicago
        - UMASK_SET=022 #optional
    volumes:
        - /srv/dev-disk-by-label-data/appdata/radarr:/config
        - /srv/dev-disk-by-label-data/media/movies:/movies
        - /srv/dev-disk-by-label-data/downloads/completed:/downloads
    restart: unless-stopped

 sonarr:
    image: linuxserver/sonarr
    network_mode: "service:transmission-openvpn"
    container_name: sonarr
    environment:
        - PUID=998
        - PGID=100
        - TZ=America/Chicago
        - UMASK_SET=022 #optional
    volumes:
        - /srv/dev-disk-by-label-data/appdata/sonarr:/config
        - /srv/dev-disk-by-label-data/media/tv:/tv
        - /srv/dev-disk-by-label-data/downloads/completed:/downloads
    restart: unless-stopped

 lidarr:
    image: linuxserver/lidarr
    network_mode: "service:transmission-openvpn"
    container_name: lidarr
    environment:
        - PUID=998
        - PGID=100
        - TZ=America/Chicago
        - UMASK_SET=022 #optional
    volumes:
        - /srv/dev-disk-by-label-data/appdata/lidarr:/config
        - /srv/dev-disk-by-label-data/media/music:/music
        - /srv/dev-disk-by-label-data/downloads/completed:/downloads
    restart: unless-stopped

Everyting else looks to be working fine but I can not open or run anything in the stack that ive added.
Ill be straight up Im completely new to everything linux, NAS, docker, portainer, etc., etct. based.

So any help to get this working would be great.
If I need to add more details as to what is going on just please feel free to ask.

**Comment by mhazy** [comment]
It would be helpful to see the log output from `docker-compose up`, specifically for the transmission service, otherwise it's impossible to know what's happening.

Also, not 100% sure if you put random stuff in yourself but it looks like you might have exposed your VPN credentials (OPENVPN_USERNAME, and OPENVPN_PASSWORD) in your config.

**Comment by st1650** [comment]
I have a similar setup and got it working by doing a manual ovpn file.
Thecus N5550 NAS running OMV 5 on x86-64
I installed docker through the web-ui but also install openvpn manually using apt install openvpn (and dependencies)
I couldn't get the docker container to run using surfshark so I I decided to manually run openvpn with a ovpn provided by surfshark and a txt with my credentials and lo and behold it worked fine. So I created a folder in /cfg and put my 2 files (udp.ovpn and openvpn-credentials.txt ) and ran the command as follow:

docker run --cap-add=NET_ADMIN -d \
              -v /srv/dev-disk-by-id-md-name-N5550-0/data/Torrent/:/data \
              -v /etc/localtime:/etc/localtime:ro \
              -v /cfg/udp.ovpn:/etc/openvpn/custom/default.ovpn \
              -v /cfg/:/etc/openvpn/custom/ \
              -e CREATE_TUN_DEVICE=true \
              -e OPENVPN_USERNAME=xxxxxxxxx(yourusername) \
              -e OPENVPN_PASSWORD=zzzzzzz(yourpwd) \
              -e OPENVPN_PROVIDER=CUSTOM \
              -e WEBPROXY_ENABLED=false \
              -e LOCAL_NETWORK=192.168.1.0/24 \
              --log-driver json-file \
              --log-opt max-size=10m \
              -p 9091:9091 \
              haugene/transmission-openvpn

Obviously the first line has to be modified for your data drive. But yeah try a manual ovpn file and see if it connects and properly assign an IP address (ifconfig -a). Also I had to disable ipv6 as well by modifying your /etc/sysctl.conf.

Disable ipv6 by adding or uncommenting these lines:

net.ipv6.conf.all.disable_ipv6 = 1

net.ipv6.conf.default.disable_ipv6 = 1

net.ipv6.conf.lo.disable_ipv6 = 1

Enable packet forwarding:

Uncomment the next line to enable packet forwarding for IPv4
net.ipv4.ip_forward=1

Good luck
